lib/rubocop/cop/style/hash_new_keyword_arguments.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Avoid passing default value to `Hash.new` as keyword arguments.
+      #
+      # @example
+      #   # bad
+      #   Hash.new(foo: 1)
+      #
+      #   # good
+      #   Hash.new({ foo: 1 })
+      class HashNewKeywordArguments < Base
+        extend AutoCorrector
+
+        MSG = 'Avoid passing default value to `Hash.new` as keyword arguments.'
+
+        RESTRICT_ON_SEND = %i[new].freeze
+
+        # @!method find_keyword_arguments_from_hash_new(node)
+        #   @param [RuboCop::AST::Node] node
+        #   @return [RuboCop::AST::HashNode, nil]
+        def_node_matcher :find_keyword_arguments_from_hash_new, <<~PATTERN
+          (send
+            (const {nil? cbase} :Hash)
+            :new
+            $#keyword_arguments?
+          )
+        PATTERN
+
+        def on_send(node)
+          keyword_arguments = find_keyword_arguments_from_hash_new(node)
+          return unless keyword_arguments
+
+          add_offense(keyword_arguments) do |corrector|
+            corrector.wrap(keyword_arguments, '{ ', ' }')
+          end
+        end
+
+        private
+
+        def keyword_arguments?(argument)
+          argument.hash_type? && !argument.source.lstrip.start_with?('{')
+        end
```

Comment:
Is there any other good way to inspect keyword arguments?

Reply:
It seems that `kwargs` is rounded to `hash` in rubocop-ast.
https://github.com/rubocop/rubocop-ast/blob/57bc6792ed83dfb2890ed1ce1eebc2580fd83aa2/lib/rubocop/ast/builder.rb#L64

I don't know if this is better, but would it be better to define a `KwargsNode` to replace it?

Reply:
Since the `emit_kwargs` option is not enabled in RuboCop's parser, I don't think the `kwargs` type is actually passed there, but changing this option might have too much impact.

- https://github.com/whitequark/parser/blob/8de8b7fa7af471a2159860d6a0a5b615eac9c83c/lib/parser/builders/default.rb#L131-L177